modules/3d/include/opencv2/3d.hpp

```diff
@@ -468,77 +468,308 @@ can be found in:
+        bool jacobiScaling;
+        // double upFactor until the probe is successful
+        bool upDouble;
+        // use stepQuality metrics for steps down
+        bool useStepQuality;
+        // clamp diagonal values added to J^T*J to pre-defined range of values
+        bool clampDiagonal;
+        // to use squared L2 norm or Inf norm for step size estimation
+        bool stepNormInf;
+        // to use relEnergyDeltaTolerance or not
+        bool checkRelEnergyChange;
+        // to use minGradientTolerance or not
+        bool checkMinGradient;
+        // to use stepNormTolerance or not
+        bool checkStepNorm;
+        // to use geodesic acceleration or not
+        bool geodesic;
+        // second directional derivative approximation step for geodesic acceleration
+        double hGeo;
+        // how much of geodesic acceleration is used
+        double geoScale;
+        // optimization stops when norm2(dx) drops below this value
+        double stepNormTolerance;
+        // optimization stops when relative energy change drops below this value
+        double relEnergyDeltaTolerance;
+        // optimization stops when max gradient value (J^T*b vector) drops below this value
+        double minGradientTolerance;
+        // optimization stops when energy drops below this value
+        double smallEnergyTolerance;
+        // optimization stops after a number of iterations performed
+        unsigned int maxIterations;
+
+        // LevMarq up and down params
+        double initialLambda;
+        double initialLmUpFactor;
+        double initialLmDownFactor;
+    };
+
+    /*
+    Defined in details header
+    */
+    class CV_EXPORTS Backend
     {
     public:
-        virtual ~Callback() {}
-        /**
-         computes error and Jacobian for the specified vector of parameters
-
-         @param param the current vector of parameters
-         @param err output vector of errors: err_i = actual_f_i - ideal_f_i
-         @param J output Jacobian: J_ij = d(err_i)/d(param_j)
-
-         when J=noArray(), it means that it does not need to be computed.
-         Dimensionality of error vector and param vector can be different.
-         The callback should explicitly allocate (with "create" method) each output array
-         (unless it's noArray()).
-        */
-        virtual bool compute(InputArray param, OutputArray err, OutputArray J) const = 0;
+        virtual ~Backend() { }
     };
 
-    /**
-       Runs Levenberg-Marquardt algorithm using the passed vector of parameters as the start point.
-       The final vector of parameters (whether the algorithm converged or not) is stored at the same
-       vector. The method returns the number of iterations used. If it's equal to the previously specified
-       maxIters, there is a big chance the algorithm did not converge.
+    LevMarqBase(const Ptr<Backend>& backend, const Settings& settings);
```

Comment:
Users don't really need this constructor. As they can't use it.

Reply:
Fixed, see the comment below